SDK_versioned_docs/version-3.0.0/guides/04-execute-a-swap.md

```diff
@@ -0,0 +1,249 @@
+        uint24 fee,
+        uint256 amountIn,
+        uint160 sqrtPriceLimitX96
+    ) external returns (uint256 amountOut);
+```
+
+In an ideal world, these quoter functions would be `view` functions, which would make them very easy to query on-chain with minimal gas costs. Instead, the V3 quoter contracts rely on state-changing calls designed to be reverted to return the desired data. This means calling the quoter will be very expensive and should not be called on-chain.
+
+To get around this difficulty, we can use the [callStatic](https://docs.ethers.io/v5/api/contract/contract/#contract-callStatic) method provided by `ethers.js`. `callStatic` is a useful method that submits a state-changing transaction to an Ethereum node, but asks the node to simulate the state change, rather than to execute it. Our script can then return the result of the simulated state change.
+
+To simulate a transaction without actually broadcasting it to the EVM, use the `callStatic` to call the `ExactInputSingle` function in the `Quoter` contract, which will tell us how much an of output token we will receive given a certain amount of input token when using a single hop swap.
+
+Note this function uses the `Immutables` interface defined in the earlier guides.
+
+```ts
+async function main() {
+  const amountIn = 1500;
+
+  const quotedAmountOut = await quoterContract.callStatic.quoteExactInputSingle(
+    immutables.token0,
+    immutables.token1,
+    immutables.fee,
+    amountIn.toString(),
+    0
+  );
+}
+```
+
+## Construct A Trade
+
+Create a [Route](https://github.com/Uniswap/uniswap-v3-sdk/blob/7c3aedd0cf9441d03607e258734eada44a73863d/src/entities/route.ts) object and assign it to a variable `swapRoute`
+
+```ts
+const swapRoute = new Route([poolExample], TokenA, TokenB);
+```
+
+Create an [Unchecked Trade](https://github.com/Uniswap/uniswap-v3-sdk/blob/7c3aedd0cf9441d03607e258734eada44a73863d/src/entities/trade.ts#L346), a type of trade that is useful when we have retrieved a quote prior to the construction of the trade object.
+
+```ts
+const uncheckedTradeExample = await Trade.createUncheckedTrade({
+  route: swapRoute,
+  inputAmount: CurrencyAmount.fromRawAmount(TokenA, amountIn.toString()),
+  outputAmount: CurrencyAmount.fromRawAmount(TokenB, quotedAmountOut.toString()),
+  tradeType: TradeType.EXACT_INPUT,
+});
+```
+
+## Print The Quote And Trade To Your Console
+
+Print the `Quote` and the `UncheckedTrade` to the console.
+
+```ts
+console.log("The quoted amount out is", quotedAmountOut.toString());
+console.log("The unchecked trade object is", uncheckedTradeExample);
+```
+
+If everything is working, you should see something similar to this returned in your console
+
+```console
+The quoted amount out is 661497830963
+The unchecked trade object is Trade {
+  swaps: [
+    {
+      inputAmount: [CurrencyAmount],
+      outputAmount: [CurrencyAmount],
+      route: [Route]
+    }
+  ],
+  tradeType: 0
+}
+```
+
+## The Full Example
```

Comment:
do we usually also link out to a github repo containing the full code?

Reply:
afaik we do for smart contract examples, but not the javascript ones